x/interchaintxs/keeper/msg_server.go

```diff
@@ -119,7 +119,7 @@ func (k Keeper) SubmitTx(goCtx context.Context, msg *ictxtypes.MsgSubmitTx) (*ic
 		Memo: msg.Memo,
 	}
 
-	timeoutTimestamp := time.Now().Add(InterchainTxTimeout).UnixNano()
+	timeoutTimestamp := ctx.BlockTime().Add(time.Duration(msg.Timeout) * time.Second).UnixNano()
```

Comment:
`InterchainTxTimeout` is dead const now.

Reply:
fixed